shared/js/content-scripts/element-hiding.js

```diff
@@ -0,0 +1,87 @@
+var isTop = window === window.top;
+var contentScript = {};
+var csPort = chrome.runtime.connect({name:"cs-port"});
+var frameId;
+var mainFrameUrl;
+var possibleTargets = [];
+var foundScripts = [];
+var foundFrames = [];
+
+
+window.addEventListener("message", (e) => {
+    if (isTop) {
+        if (e.data.frameUrl) {
```

Comment:
Would it make sense to pass some data.type in the message instead of checking for frameUrl or frameId? `data.type === 'hide element'` That might make it easier to follow what each message is doing.